src/test/objectstore/store_test.cc

```diff
@@ -7259,7 +7259,8 @@ TEST_P(StoreTestSpecificAUSize, BlobReuseOnOverwrite) {
 }
 
 TEST_P(StoreTestSpecificAUSize, ZeroBlockDetectionSmallAppend) {
-  if (string(GetParam()) != "bluestore") {
+  CephContext *cct = (new CephContext(CEPH_ENTITY_TYPE_CLIENT))->get();
+  if (string(GetParam()) != "bluestore" || !cct->_conf->bluestore_zero_block_detection) {
     return;
```

Comment:
I would use `GTEST_SKIP` here, to avoid the appearance of a passing test case when it is not actually run.  Something like:
```
GTEST_SKIP() << "not bluestore or bluestore_zero_block_detection=false, skipping";
```

Reply:
Cleaned up the commits a bit. I grouped all of the essential BlueStore commits together and created two separate commits: one for the QA suite that enables this feature for Teuthology, and another that uses GTEST_SKIP for the smr check as well, to similarly not produce false results if the test gets skipped due to detecting smr.